### PR TITLE
Rework minimaps

### PR DESCRIPTION
**Pull request by neersighted**
box station_1
Minimaps now only generate for Z1, and generated based on a md5sum of the main z-level. In addition, they show many more objects and the resolution has been increased to 16x16 tiles.

### CODE REVIEW

File: code/controllers/subsystem/minimap.dm [hunk 1]
```diff
@@ -21,163 +15,77 @@ var/datum/subsystem/minimap/SSminimap
...
-#undef MINIMAP_DEBUG
-#endif
+/datum/subsystem/minimap/proc/generate(z = 1, x1 = 1, y1 = 1, x2 = world.maxx, y2 = world.maxy)
+	// Load the background.
+	var/icon/minimap = new /icon('icons/minimap.dmi')
+	// Scale it up to our target size.
+	minimap.Scale(MINIMAP_SIZE, MINIMAP_SIZE)
+
+	// Loop over turfs and generate icons.
+	for(var/turf/tile in block(locate(x1, y1, z), locate(x2, y2, z)))
```
Comment (RemieRichards) [inline code/controllers/subsystem/minimap.dm]: ```block()``` returns only turfs, istypeless-forloop please

File: code/controllers/subsystem/minimap.dm [hunk 2]
```diff
@@ -21,163 +15,77 @@ var/datum/subsystem/minimap/SSminimap
...
+			if(obj)
+				obj_icons += getFlatIcon(obj)
+			obj = locate(/obj/machinery) in tile
+			if(obj)
+				obj_icons += new /icon(obj.icon, obj.icon_state, obj.dir, 1, 0)
+			obj = locate(/obj/structure/window) in tile
+			if(obj)
+				obj_icons += new /icon('icons/obj/smooth_structures/window.dmi', "window", SOUTH)
+
+			for(var/icon/obj_icon in obj_icons)
```
Comment (RemieRichards) [inline code/controllers/subsystem/minimap.dm]: this is a list of ```/icon```s, istypeless-forloop please

**Comment by PKPenguin321** [comment]
performance is better or worse (not counting performance increase from only rendering z1)?

**Comment by neersighted** [comment]
16 seconds initial; 0 seconds until next map edit.

**Comment by WJohn** [comment]
Sounds around the same as what we have now. Minimaps have always been pretty slow. I do wish we made use of it more, though.

@neersighted, maybe we could use them in power and atmos monitoring consoles to highlight the affected areas?

**Comment by neersighted** [comment]
Sure, I'll see about adding them later, after I rewrote crew monitors.

**Comment by neersighted** [comment]
@WJohn the big difference is 0 seconds every round after initial, where as it takes about 5-6 seconds most rounds right now (to checksum).

File: code/controllers/subsystem/minimap.dm [hunk 3]
```diff
@@ -1,19 +1,13 @@
...
 	name = "Minimap"
 	priority = -2
 
-	var/const/MAX_ICON_DIMENSION = 1024
-	var/const/ICON_SIZE = 4
+	var/const/MINIMAP_SIZE = 2048
+	var/const/TILE_SIZE = 8
 
-	var/max_initalized_zlevel = 0
+	var/list/z_levels = list(1)
```
Comment (tkdrg) [inline code/controllers/subsystem/minimap.dm]: `ZLEVEL_STATION`

**Pull request merged**